From 41165cc2bdae9c522bae8128bae915ce19fc587e Mon Sep 17 00:00:00 2001
From: Sandeep Gutta <sangutta@codeaurora.org>
Date: Fri, 12 Sep 2014 01:14:35 +0530
Subject: [PATCH 1/4] MSIM: Phone account related changes

- Do not set "always ask" if only one account present
- Use voice subId from SIM Settings as default outgoing account

Change-Id: I9e7b16f20c357f8d3ceedfba781bc893d11c39ea
---
 .../server/telecom/PhoneAccountRegistrar.java      | 30 ++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/src/com/android/server/telecom/PhoneAccountRegistrar.java b/src/com/android/server/telecom/PhoneAccountRegistrar.java
index 7090c21..11efa66 100644
--- a/src/com/android/server/telecom/PhoneAccountRegistrar.java
+++ b/src/com/android/server/telecom/PhoneAccountRegistrar.java
@@ -232,6 +232,10 @@ public class PhoneAccountRegistrar {
         DefaultPhoneAccountHandle defaultPhoneAccountHandle = mState.defaultOutgoingAccountHandles
                 .get(userHandle);
         if (defaultPhoneAccountHandle == null) {
+            if (TelephonyManager.getDefault().getPhoneCount() > 1 &&
+                       mSubscriptionManager.getActiveSubscriptionInfoCount() == 1) {
+                return getUserSelectedVoicePhoneAccount();
+            }
             return null;
         }
         // Make sure the account is still registered and owned by the user.
@@ -244,6 +248,32 @@ public class PhoneAccountRegistrar {
         return null;
     }
 
+    PhoneAccountHandle getUserSelectedVoicePhoneAccount() {
+        long voiceSubId = SubscriptionManager.getDefaultVoiceSubscriptionId();
+        PhoneAccountHandle prefPhoneAccount = null;
+
+        Log.i(this, "getUserSelVoicePhoneAccount, voice subId = " + voiceSubId);
+        for (int i = 0; i < mState.accounts.size(); i++) {
+            String id = mState.accounts.get(i).getAccountHandle().getId();
+
+            // emergency account present return it
+            if (id.equals("E")) {
+                Log.i(this, "getUserSelVoicePhoneAccount, emergency account ");
+                return mState.accounts.get(i).getAccountHandle();
+            }
+
+            long subId = Long.parseLong(id);
+            Log.i(this, "getUserSelectedVoicePhoneAccount, voice subId = "
+                         + voiceSubId + " subId = " + subId + " mId = " + id);
+            if (subId == voiceSubId) {
+                prefPhoneAccount = mState.accounts.get(i).getAccountHandle();
+                break;
+            }
+        }
+
+        return prefPhoneAccount;
+    }
+
     /**
      * Sets the phone account with which to place all calls by default. Set by the user
      * within phone settings.
-- 
2.7.4

